src/main/java/com/formulasearchengine/mathmlquerygenerator/XQueryGenerator.java

```diff
@@ -117,53 +132,73 @@ public void setMainElement( Node mainElement ) {
 		lengthConstraint = "";
 	}
 
+	/**
+	 * Generates the constraints of the XQuery and then builds the XQuery and returns it as a string
+	 * @return XQuery as string
+	 */
 	public String toString() {
 		if ( mainElement == null ) {
 			return null;
 		}
-		String fixedConstraints = generateConstraint( mainElement, true );
-		String qvarConstraintString = "";
+		final String fixedConstraints = generateConstraints( mainElement, true );
+		final StringBuilder qvarConstraintString = new StringBuilder();
+		//Generate qvar constraint string
+		//This specifies that the same qvars must refer to the same nodes, using the XQuery "=" equality
+		//This is equality based on: same text, same node names, and same children by the "=" equality
 		for ( Map.Entry<String, ArrayList<String>> entry : qvar.entrySet() ) {
-			String addString = "";
-			boolean newContent = false;
+			final StringBuilder addString = new StringBuilder();
 			if ( entry.getValue().size() > 1 ) {
-				String first = entry.getValue().get( 0 );
-				if ( qvarConstraintString.length() > 0 ) {
-					addString += "\n  and ";
+				final String firstEntry = entry.getValue().get( 0 );
+				if ( qvarConstraintString.length() != 0 ) {
```

Comment:
@physikerwelt It's a StringBuilder object, which doesn't have isEmpty() for some reason